lib/src/layer.dart

```diff
@@ -1,128 +1,233 @@
 part of tiled;
 
 class Layer {
-  static const int FLIPPED_HORIZONTALLY_FLAG = 0x80000000;
-  static const int FLIPPED_VERTICALLY_FLAG = 0x40000000;
-  static const int FLIPPED_DIAGONALLY_FLAG = 0x20000000;
-
-  String name;
-  int width;
+  List<Chunk> chunks = [];
+  String
+      compression; // zlib, gzip, zstd (since Tiled 1.3) or empty (default). tilelayer
+  List<int> data = [];
+  String draworder = 'topdown'; //topdown (default) or index. only objectgroup
+  String color; // only objectgroup; Not supported by json
+  String encoding = 'csv'; // csv (default) or base64. tilelayer
   int height;
+  int id;
+  TiledImage image; // only on imageLayer
+  List<Layer> layers = [];
+  String name;
+  List<TiledObject> objects = [];
+  double offsetx;
+  double offsety;
+  double opacity;
+  List<Property> properties = [];
+  int startx;
+  int starty;
+  String tintcolor;
+  String transparentcolor;
+  String type; // tilelayer, objectgroup, imagelayer or group
```

Comment:
should this be an enum?

Reply:
enums are a pain. Or i am too stupid.
Example:
enum FileEncoding { csv, base64 }

Json Parsing via:
    encoding = FileEncoding.values.firstWhere(
        (e) => e.toString() ==json['encoding'],
        orElse: () => null);
fails, because e.toString() is  "FileEncoding.base64" instead of "base64"

Only for this i had to add a new dependency to package:flutter/foundation.dart to use something like this:
describeEnum(e) == json['encoding']

I am not sure, but i dont think this dependency is worth it.
Enums in dart are not my favorite. Like them in java, but not in dart. Or i have not understood how this should work....

Reply:
ok, i did it with extensions. But for that i had to change the sdk-version from 
  sdk: ">=2.3.0 <3.0.0"
to
  sdk: ">=2.7.0 <3.0.0"

If have a better idea. Just say it.

Reply:
oh I love it, that is very clever! and the sdk constraint is fine, 2.7 is very stable and flame and flutter all have the 2.7 requirement now